librarian/data/manager.py

```diff
@@ -1,180 +1,256 @@
+import itertools
 import os
 import re
-import mimetypes
-from itertools import izip_longest
 
-from bottle import request
-from bottle_utils.common import to_unicode
+from ..core.exts import ext_container as exts
+from .meta.archive import Archive
 
-from .dirinfo import DirInfo
-from .facets.utils import get_facets, get_dir_facets, get_archive
 
+class Error(Exception):
+    """
+    Base class for all :py:class:`Manager` related exceptions.
+    """
+    pass
 
-# Match any string that starts with a period, or has at least one path
-# separator followed by a period
-RE_PATH_WITH_HIDDEN = re.compile(r'^(\.|.*{}\.)'.format(os.sep))
 
-
-def nohidden(fsos):
-    for fso in fsos:
-        if not RE_PATH_WITH_HIDDEN.match(fso.rel_path):
-            yield fso
-
-
-def unique_results(*iterables):
-    results = set()
-    for it in iterables:
-        results |= set(it)
-    return list(results)
+class InvalidQuery(Error):
+    """
+    Raised when an invalid path or search query is passed to one of the query
+    methods on :py:class:`Manager`.
+    """
+    pass
 
 
 class Manager(object):
+    """
+    Provides a facade over various lower level APIs, such as the file system
+    abstraction layer, py:class:`Facet` and py:class:`DirInfo` facilities. The
```

Comment:
This comment could be confusing in the future since Facet & DirInfo classes do not exist anymore